pkg/runctl/run_manager_test.go

```diff
@@ -1767,6 +1816,155 @@ func Test_RunManager_Log_Elasticsearch(t *testing.T) {
+	 */
+	test = "ErrorOnWrongFormattedIndexURL"
+	for _, tc := range []struct {
+		name         string
+		indexURLJson string
+	}{
+		{"indexURLWithNoScheme", `"indexURL": "testURL"`},
+		{"indexURLWithWrongFormat2", `"indexURL": "http//testURL"`},
+	} {
+		t.Run(test+"_"+tc.name, func(t *testing.T) {
+			var err error
+
+			// setup
+			pipelineRunJSON := fmt.Sprintf(fixIndent(`
+				 {
+					 "apiVersion": "steward.sap.com/v1alpha1",
+					 "kind": "PipelineRun",
+					 "metadata": {
+						 "name": "dummy1",
+						 "namespace": "namespace1"
+					 },
+					 "spec": {
+						 "jenkinsFile": {
+							 "repoUrl": "dummyRepoUrl",
+							 "revision": "dummyRevision",
+							 "relativePath": "dummyRelativePath"
+						 },
+						 "logging": {
+							 "elasticsearch": {
+								 %s
+							 }
+						 }
+					 }
+				 }`),
+				tc.indexURLJson,
+			)
+			t.Log("input:", pipelineRunJSON)
+			examinee, runCtx, _ := setupExaminee(t, pipelineRunJSON)
+			// exercise
+			err = examinee.createTektonTaskRun(runCtx)
+			assert.ErrorContains(t, err, "field spec.logging.elasticSearch.indexURL is invalid: scheme not supported")
+		})
+	}
+
+	/**
+	 * Test: If provided indexURL at spec.logging.elasticsearch.indexURL
+	 * does not have correct scheme tests should fail.
+	 */
+	test = "ErrorOnWrongSchemeInIndexURL"
+	for _, tc := range []struct {
+		name         string
+		indexURLJson string
+	}{
+		{"indexURLWithIncorrectScheme", `"indexURL": "ftp://testURL"`},
+	} {
+		t.Run(test+"_"+tc.name, func(t *testing.T) {
+			var err error
+
+			// setup
+			pipelineRunJSON := fmt.Sprintf(fixIndent(`
+				  {
+					  "apiVersion": "steward.sap.com/v1alpha1",
+					  "kind": "PipelineRun",
+					  "metadata": {
+						  "name": "dummy1",
+						  "namespace": "namespace1"
+					  },
+					  "spec": {
+						  "jenkinsFile": {
+							  "repoUrl": "dummyRepoUrl",
+							  "revision": "dummyRevision",
+							  "relativePath": "dummyRelativePath"
+						  },
+						  "logging": {
+							  "elasticsearch": {
+								  %s
+							  }
+						  }
+					  }
+				  }`),
+				tc.indexURLJson,
+			)
+			t.Log("input:", pipelineRunJSON)
+			examinee, runCtx, _ := setupExaminee(t, pipelineRunJSON)
+			// exercise
+			err = examinee.createTektonTaskRun(runCtx)
+			assert.ErrorContains(t, err, "field spec.logging.elasticSearch.indexURL is invalid: scheme not supported")
+		})
+	}
+
+	/**
+	 * Test: Test should pass if indexURL of elasticSearch has correct format and scheme.
+	 */
+	test = "CorrectFormatForIndexURL"
+	for _, tc := range []struct {
+		name               string
+		indexURLJson       string
+		expectedParamValue string
+	}{
+		{"validhttpURL", `"indexURL": "http://host.domain"`, "http://host.domain"},
+		{"validhttpsURL", `"indexURL": "https://host.domain"`, "https://host.domain"},
+		{"validHTTPURL", `"indexURL": "HTTP://host.domain"`, "http://host.domain"},
+		{"validHTTPSURL", `"indexURL": "HTTPS://host.domain"`, "https://host.domain"},
```

Comment:
You can reduce field `indexURLJson` to just the URL string.

Reply:
done